bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQBridgeHandler.java

```diff
@@ -0,0 +1,375 @@
+        lGDeviceRegister.remove(thing.getDeviceId());
+    }
+
+    private final LGDevicePollingRunnable lgDevicePollingRunnable;
+
+    private LGThinQGeneralApiClientService getLgApiClient() {
+        return Objects.requireNonNull(lgApiClient, "Not expected lgApiClient null. It most likely a bug");
+    }
+
+    class LGDevicePollingRunnable extends PollingRunnable {
+        public LGDevicePollingRunnable(String bridgeName) {
+            super(bridgeName);
+        }
+
+        @Override
+        protected void doConnectedRun() throws LGThinqException {
+            Map<String, LGDevice> lastDevicesDiscoveredCopy = new HashMap<>(lastDevicesDiscovered);
+            List<LGDevice> devices = getLgApiClient().listAccountDevices(bridgeName);
+            // if not registered yet, and not discovered before, then add to discovery list.
+            devices.forEach(device -> {
+                String deviceId = device.getDeviceId();
+                logger.debug("Device found: {}", deviceId);
+                if (lGDeviceRegister.get(deviceId) == null && !lastDevicesDiscovered.containsKey(deviceId)) {
+                    logger.debug("Adding new LG Device to things registry with id:{}", deviceId);
+                    if (discoveryService != DUMMY_DISCOVERY_SERVICE) {
+                        discoveryService.addLgDeviceDiscovery(device);
+                    }
+                } else {
+                    if (discoveryService != DUMMY_DISCOVERY_SERVICE && lGDeviceRegister.get(deviceId) != null) {
+                        discoveryService.removeLgDeviceDiscovery(device);
+                    }
+                }
+                lastDevicesDiscovered.put(deviceId, device);
+                lastDevicesDiscoveredCopy.remove(deviceId);
+            });
+            // the rest in lastDevicesDiscoveredCopy is not more registered in LG API. Remove from discovery
+            lastDevicesDiscoveredCopy.forEach((deviceId, device) -> {
+                logger.debug("LG Device '{}' removed.", deviceId);
+                lastDevicesDiscovered.remove(deviceId);
+
+                LGThinQAbstractDeviceHandler<? extends CapabilityDefinition, ? extends SnapshotDefinition> deviceThing = lGDeviceRegister
+                        .get(deviceId);
+                if (deviceThing != null) {
+                    deviceThing.onDeviceRemoved();
+                }
+                if (discoveryService != DUMMY_DISCOVERY_SERVICE && deviceThing != null) {
+                    discoveryService.removeLgDeviceDiscovery(device);
+                }
+            });
+
+            lGDeviceRegister.values().forEach(LGThinQAbstractDeviceHandler::refreshStatus);
+        }
+    }
+
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        List<ConfigStatusMessage> resultList = new ArrayList<>();
+        if (lgthinqConfig.username.isEmpty()) {
```

Comment:
Minor improvement, same for other configuration keys
```suggestion
        if (lgthinqConfig.username.isBlank()) {
```